pygmt/helpers/utils.py

```diff
@@ -324,11 +426,22 @@ def build_arg_list(
         elif value is True:
             gmt_args.append(f"-{key}")
         elif is_nonstr_iter(value):
-            gmt_args.extend(non_ascii_to_octal(f"-{key}{_value}") for _value in value)
+            gmt_args.extend(f"-{key}{_value}" for _value in value)
         else:
-            gmt_args.append(non_ascii_to_octal(f"-{key}{value}"))
+            gmt_args.append(f"-{key}{value}")
+
+    # Convert non-ASCII characters (if any) in the arguments to octal codes
+    encoding = _check_encoding("".join(gmt_args))
+    if encoding != "ascii":
+        gmt_args = [non_ascii_to_octal(arg, encoding=encoding) for arg in gmt_args]
```

Comment:
This block assumes that there will only be one encoding type? To support multiple encodings in different arguments, maybe change it to something like this (untested):

```suggestion
    # Convert non-ASCII characters (if any) in the arguments to octal codes
    for i, arg in enumerate(gmt_args):
        encoding = _check_encoding("".join(gmt_args))
        if encoding != "ascii":
            gmt_args[i] = non_ascii_to_octal(arg, encoding=encoding)
```

But this can be done in a separate PR perhaps. I'm not sure how common it is to mix encodings in different arguments.

Reply:
> To support multiple encodings in different arguments, maybe change it to something like this (untested):

It's not possible in GMT. For each GMT module call, we can only pass `--PS_CHAR_ENCODING` once, so it means that all the arguments in a GMT call must be in the same encoding. 

For `Figure.text`, the `text` string and any arguments also must use the same encoding.

Reply:
If people use mixed encodings in different arguments, the default `ISOLatin1+` encoding is used.